queue_services/entity-digital-credentials/src/entity_digital_credentials/worker.py

```diff
@@ -0,0 +1,142 @@
+# Copyright © 2019 Province of British Columbia
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""The unique worker functionality for this service is contained here.
+
+The entry-point is the **cb_subscription_handler**
+
+The design and flow leverage a few constraints that are placed upon it
+by NATS Streaming and using AWAIT on the default loop.
+- NATS streaming queues require one message to be processed at a time.
+- AWAIT on the default loop effectively runs synchronously
+
+If these constraints change, the use of Flask-SQLAlchemy would need to change.
+Flask-SQLAlchemy currently allows the base model to be changed, or reworking
+the model to a standalone SQLAlchemy usage with an async engine would need
+to be pursued.
+"""
+import json
+import os
+
+import nats
+from entity_queue_common.service import QueueServiceManager
+from entity_queue_common.service_utils import QueueException, logger
+from flask import Flask
+from legal_api import db
+from legal_api.core import Filing as FilingCore
+from legal_api.models import Business
+from legal_api.services import digital_credentials, flags
+from sqlalchemy.exc import OperationalError
+
+from entity_digital_credentials import config
+from entity_digital_credentials.digital_credentials_processors import (
+    business_number,
+    change_of_registration,
+    dissolution,
+    manual,
+    put_back_on,
+)
+
+
+qsm = QueueServiceManager()  # pylint: disable=invalid-name
+
+APP_CONFIG = config.get_named_config(os.getenv('DEPLOYMENT_ENV', 'production'))
+FLASK_APP = Flask(__name__)
+FLASK_APP.config.from_object(APP_CONFIG)
+db.init_app(FLASK_APP)
+
+with FLASK_APP.app_context():  # db require app context
+    digital_credentials.init_app(FLASK_APP)
+
+if FLASK_APP.config.get('LD_SDK_KEY', None):
+    flags.init_app(FLASK_APP)
+
+
+async def process_digital_credential(dc_msg: dict, flask_app: Flask):
+    # pylint: disable=too-many-branches, too-many-statements
+    """Process any digital credential messages in queue."""
+    if not flask_app:
+        raise QueueException('Flask App not available.')
+
+    with flask_app.app_context():
+        logger.debug('Attempting to process digital credential message: %s', dc_msg)
+
+        if dc_msg is None:
+            raise QueueException
+
+        if dc_msg['type'] is None:
+            raise QueueException('Digital credential message is missing type.')
```

Comment:
I think this would go away if we add the upfront check that entity-bn does that I mentioned above.

Also would it make sense to do a check the business type here and return if the legal type isn't SP?

Reply:
> Also would it make sense to do a check the business type here and return if the legal type isn't SP?

It's SP only during this pilot but may be extended later. Is there any to not bake this into code? Use config instead?

Reply:
yeah maybe a config.  Still thinking about this one.

Reply:
We can address the legal type in a subsequent PR if we end up deciding to filter by legal type.  in the meantime, a DC just won't be found and a warning is logged.